scripts/generate_config_tests.py

```diff
@@ -87,7 +87,7 @@ def dependencies_of_setting(cfg: config.Config,
         if name.startswith('MBEDTLS_CIPHER_PADDING_'):
             return 'MBEDTLS_CIPHER_C:MBEDTLS_CIPHER_MODE_CBC'
         if name.startswith('MBEDTLS_PK_PARSE_EC_'):
-            return 'MBEDTLS_PK_C:MBEDTLS_PK_HAVE_ECC_KEYS'
+            return 'MBEDTLS_PK_C:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'
```

Comment:
We need this change for development (to be able to merge #9492) and to keep `MBEDTLS_PK_HAVE_ECC_KEYS` for 3.6. There has been some discussion how to do that (see #51) but to unblock #9492 let's go with a local ad-hoc solution here based on build_tree.is_mbedtls_3_6(). Something like (not tested):
```py
if build_tree.is_mbedtls_3_6():
    return 'MBEDTLS_PK_C:MBEDTLS_PK_HAVE_ECC_KEYS'
else:
    return 'MBEDTLS_PK_C:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'
```
with a comment indicating that it is temporary and pointing to #51.

Otherwise, let's do the same in #31 and eventually update #9492 with both #31 and #34 changes in the framework.